src/component/MessageComponent.js

```diff
@@ -1,27 +1,25 @@
-import React,  { useEffect }  from 'react';
+import React, { useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 import { getMessage } from '../redux/messages/messageSlice';
 import AllMessage from './AllMessage';
 
 const MessageComponent = () => {
+  const allMessages = useSelector((state) => state.msg);
+  const dispatch = useDispatch();
 
-    const all_messages = useSelector((state) => state.msg);
-    const dispatch = useDispatch();
+  useEffect(() => {
+    if (!allMessages.length) {
+      dispatch(getMessage());
+    }
+  }, [allMessages.length, dispatch]);
 
-      useEffect(() => {
-        if (!all_messages.length) {
-          dispatch(getMessage());
-        }
-      }, [all_messages.length, dispatch]);
-    
-      return (
-        <div>
-          {
-          all_messages.length ? <AllMessage msg={all_messages} /> : <h2>No Message Found</h2>
+  return (
+    <div>
+      {
+          allMessages.length ? <AllMessage msg={allMessages} /> : <h2>No Message Found</h2>
```

Comment:
- [ ] When I start the front-end server, I am greeted with the "No message found" which was strange because I tested the API with my own data and it was working perfectly.

![image](https://user-images.githubusercontent.com/86318284/210856924-75f0a2cc-77c8-426d-8762-318ee898226c.png)

On closer inspection in the networks tab, I realize that the reason why there is no data is that your API is prohibiting data from being sent to the front end because of [CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS). Luckily for us, there is a solution to solving this using the rack-cors gem ```gem 'rack-cors'```. Kindly visit the [official docs](https://github.com/cyu/rack-cors) to learn how to set it up. It is a two-step process trust me.

![image](https://user-images.githubusercontent.com/86318284/210857575-994f0361-aaac-4efc-9bdd-fde2609220de.png)